controllers/vrg_volrep.go

```diff
@@ -1271,17 +1276,52 @@ func (v *VRGInstance) selectVolumeReplicationClass(
 
 		// ReplicationClass that matches both VRG schedule and pvc provisioner
 		if schedulingInterval == v.instance.Spec.Async.SchedulingInterval {
-			v.log.Info(fmt.Sprintf("Found VolumeReplicationClass that matches provisioner and schedule %s/%s",
-				storageClass.Provisioner, v.instance.Spec.Async.SchedulingInterval))
+			matchingReplicationClassList = append(matchingReplicationClassList, replicationClass)
+		}
+	}
+
+	return v.filterDefaultVRC(matchingReplicationClassList, storageClass.Provisioner)
+}
+
+// filterDefaultVRC filters the VRC list to return VRCs with default annotation
+// if the list contains moret then one VRC.
+func (v *VRGInstance) filterDefaultVRC(
+	replicationClassList []*volrep.VolumeReplicationClass,
+	provisioner string,
+) (*volrep.VolumeReplicationClass, error) {
+	defaultReplicationClassList := []*volrep.VolumeReplicationClass{}
+
+	if len(replicationClassList) > 1 {
+		v.log.Info("Found multiple matching VolumeReplicationClasses, filtering with default annotation")
 
-			return replicationClass, nil
+		for index := range replicationClassList {
+			if replicationClassList[index].Annotations[defaultVRCAnnotationKey] == "true" {
+				defaultReplicationClassList = append(
+					defaultReplicationClassList,
+					replicationClassList[index])
+			}
 		}
+	} else {
+		defaultReplicationClassList = replicationClassList
+	}
```

Comment:
Based on the name, I expect filterDefaultVRC to accept list with more than one vrc, and return a list with only the matching vrcs based the default annotation. The caller can do the handling (0, 1, >1). This will keep this function simpler, and we will cal it only when we find multiple matching vrcs.

The flow can be:

     matchingVRCs :=  v.selectVolumeReplicationClass()

     if len(matchingVRCs) > 1 {
         matchingVRCs = filterDefaultVRC(matchingVRCs)
     }

Now we can succeed if we have have list of one vrc, or fail in all other cases.